include/mbedtls/config.h

```diff
@@ -1350,6 +1350,19 @@
  */
 #define MBEDTLS_SSL_ALL_ALERT_MESSAGES
 
+/**
+ * \def MBEDTLS_SSL_RECORD_CHECKING
+ *
+ * Enable the API mbedtls_ssl_check_record() which allows to check the
+ * validity, freshness and authenticity of an incoming record without
+ * modifying the externally visible state of the SSL context.
+ *
+ * See mbedtls_ssl_check_record() for more information.
+ *
+ * Uncomment to enable support for record checking.
+ */
+#define MBEDTLS_SSL_RECORD_CHECKING
```

Comment:
This function actually controls just the presence of an API-function and even the presence of the configuration was asked by @mpg and answered by @hanno-arm at https://github.com/ARMmbed/mbedtls-restricted/pull/618.

Reply:
Ok - I see. If it's on and unused, it will be garbage collected by the linker and so no overhead or impact on anything else.

I'm fine with it being on by default given we *could* remove the option later and just always have it on.